airsonic-main/src/main/java/org/airsonic/player/service/MediaScannerService.java

```diff
@@ -307,63 +362,58 @@ private void updateAlbum(MediaFile file, MusicFolder musicFolder, Instant lastSc
             album.setCoverArtPath(parent.getCoverArtPath());
         }
 
-        boolean firstEncounter = !lastScanned.equals(album.getLastScanned());
-        if (firstEncounter) {
+        if (firstEncounter.get()) {
             album.setFolderId(musicFolder.getId());
-            album.setDurationSeconds(0);
-            album.setSongCount(0);
-            Integer n = albumCount.get(artist);
-            albumCount.put(artist, n == null ? 1 : n + 1);
-        }
-        if (file.getDurationSeconds() != null) {
-            album.setDurationSeconds(album.getDurationSeconds() + file.getDurationSeconds());
-        }
-        if (file.isAudio()) {
-            album.setSongCount(album.getSongCount() + 1);
-        }
-        album.setLastScanned(lastScanned);
-        album.setPresent(true);
-        albumDao.createOrUpdateAlbum(album);
-        if (firstEncounter) {
+            albumCount.computeIfAbsent(artist, k -> new AtomicInteger(0)).incrementAndGet();
```

Comment:
atomic update

Reply:
Can't this be `AtomicInteger(1);` instead of `AtomicInteger(0)).incrementAndGet();`?

Reply:
No. You're misplacing the parentheses on the `computeIfAbsent()` which give a hint about the effective purpose.
- `.computeIfAbsent(k -> new Value())`
  - if value is present for key `k`, returns it
  - if value isn't present, uses the function to create a new value

So `AtomicInteger(1)` will always set the albumCount to be 1. We don't want to set it to 1. We want to increment the _existing_ count by 1. If it already exists in the map, we retrieve it and increment by 1. If it doesn't exist, then _we create it_, and increment by 1.